packages/ui/src/forum/components/Thread/ThreadTitle.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import styled from 'styled-components'
+import * as Yup from 'yup'
+
+import { InputComponent, InputText } from '@/common/components/forms'
+import { CrossIcon, VerifiedMemberIcon } from '@/common/components/icons'
+import { EditSymbol } from '@/common/components/icons/symbols'
+import { PageTitle } from '@/common/components/page/PageTitle'
+import { useApi } from '@/common/hooks/useApi'
+import { useForm } from '@/common/hooks/useForm'
+import { useModal } from '@/common/hooks/useModal'
+import { EditThreadTitleModalCall } from '@/forum/modals/EditThreadTitleModal'
+import { ForumThreadWithDetails } from '@/forum/types'
+import { useMember } from '@/memberships/hooks/useMembership'
+import { useMyMemberships } from '@/memberships/hooks/useMyMemberships'
+
+interface ThreadTitleProps {
+  thread: ForumThreadWithDetails
+}
+
+interface TitleFormFields {
+  title: string
+}
+
+const FormSchema = Yup.object().shape({})
+
+export const ThreadTitle = ({ thread }: ThreadTitleProps) => {
+  const { api, connectionState } = useApi()
+  const { member: threadAuthor } = useMember(thread.authorId)
+  const { members: myMembers } = useMyMemberships()
+  const [isEditTitle, setEditTitle] = useState<boolean>(false)
+  const { showModal, hideModal } = useModal<EditThreadTitleModalCall>()
+
+  const isMyThread = thread && myMembers.find((member) => member.id === thread.id)
+
+  const formInitializer: TitleFormFields = {
+    title: thread.title,
+  }
+  const { fields, changeField } = useForm<TitleFormFields>(formInitializer, FormSchema)
+
+  const transaction = useMemo(() => {
+    if (isMyThread && threadAuthor && connectionState === 'connected' && api) {
+      return api.tx.forum.editThreadTitle(threadAuthor.id, thread.categoryId, thread.id, fields.title)
+    }
+  }, [fields.title, threadAuthor, isMyThread, connectionState])
```

Comment:
Personally, I'd instantiate the transaction in the modal instead, and keep this component relatively light (here it will be created even for non-OP as well). It feels like a variable that belongs in the modal where the transaction takes place

Reply:
Nice catch, done ✅